src/lib.rs

```diff
@@ -267,14 +360,14 @@ impl Cascade {
 
     fn child_layer_from_bytes<R: Read>(
         mut reader: R,
-        salt: Option<Vec<u8>>,
+        salt: Vec<u8>,
         inverted: bool,
     ) -> Result<Option<Box<Cascade>>, Error> {
         let filter = match Bloom::read(&mut reader)? {
             Some(filter) => filter,
             None => return Ok(None),
         };
-        let our_salt = salt.as_ref().cloned();
+        let our_salt = salt.to_vec();
```

Comment:
Since `salt` is no longer an `Option`, handling it this way feels a little backwards - I think it would be more idiomatic for `salt` to either be a reference to the vector or just a slice that we `to_vec()` for each new `Cascade`. Not a huge deal, though.